README.md

```diff
@@ -128,6 +128,109 @@ NOTE: You can leave that static server running in one session while you continue
+remote Python debugger based on pdb.
+
+So, given you have already deployed your model and you would like to debug the
+`config-changed` hook, you should do the following:
+
+Given `prometheus` is an application name and `lma` is a Juju model name, 
+execute the following command to grab an interactive Bash shell to the operator
+pod:
+
+
+    kubectl exec -it pod/prometheus-operator-0 -n lma -- /bin/sh
+
+
+Install the editor and the RPDB:
+
+
+    apt update
+    apt install telnet vim -y
+    pip3 install rpdb
+
+
+Open the charm entry point in the editor:
+
+
+    vim /var/lib/juju/agents/unit-prometheus-0/charm/src/charm.py
+
+
+Find a `on_config_changed_handler` function definition in the `charm.py` file.
+Modify it as follows:
+
+
+    def on_config_changed_handler(event, fw_adapter):
+        import rpdb
+        rpdb.set_trace()
+        # < ... rest of the code ... >
+
+
+Save the file (`:wq`). Do not close the current shell session!
+
+Open another terminal session to the machine with Juju client and trigger the
+`config-changed` hook:
+
+
+    juju config prometheus external-labels='{"foo": "bar"}'
+
+
+Do a `juju status`, until you will see the following:
+
+
+    Unit           Workload  Agent      Address    Ports     Message
+    prometheus/0*  active    executing  10.1.28.2  9090/TCP  (config-changed)
+
+This message means, that unit has started the `config-changed` hook routine and
+it was already intercepted by the rpdb.
+
+Now, return back to the operator pod session.
+
+Enter the interactive debugger:
+
+
+    telnet localhost 4444
+
+
+You should see the debugger interactive console.
+
+
+    # telnet localhost 4444
+    Trying ::1...
+    Trying 127.0.0.1...
+    Connected to localhost.
+    Escape character is '^]'.
+    > /var/lib/juju/agents/unit-prometheus-0/charm/hooks/config-changed(91)on_config_changed_handler()
+    -> set_juju_pod_spec(fw_adapter)
+    (Pdb) where
+      /var/lib/juju/agents/unit-prometheus-0/charm/hooks/config-changed(141)<module>()
+    -> main(Charm)
+      /var/lib/juju/agents/application-prometheus/charm/lib/ops/main.py(212)main()
+    -> _emit_charm_event(charm, juju_event_name)
+      /var/lib/juju/agents/application-prometheus/charm/lib/ops/main.py(128)_emit_charm_event()
+    -> event_to_emit.emit(*args, **kwargs)
+      /var/lib/juju/agents/application-prometheus/charm/lib/ops/framework.py(205)emit()
+    -> framework._emit(event)
+      /var/lib/juju/agents/application-prometheus/charm/lib/ops/framework.py(710)_emit()
+    -> self._reemit(event_path)
+      /var/lib/juju/agents/application-prometheus/charm/lib/ops/framework.py(745)_reemit()
+    -> custom_handler(event)
+      /var/lib/juju/agents/unit-prometheus-0/charm/hooks/config-changed(68)on_config_changed()
+    -> on_config_changed_handler(event, self.fw_adapter)
+    > /var/lib/juju/agents/unit-prometheus-0/charm/hooks/config-changed(91)on_config_changed_handler()
+    -> set_juju_pod_spec(fw_adapter)
+    (Pdb)
+
+
```

Comment:
From this point forward, the usual `pdb` commands apply.  For more information on how to use `pdb`, see the official [pdb documentation](https://docs.python.org/3/library/pdb.html).